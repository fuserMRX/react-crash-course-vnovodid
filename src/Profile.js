import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchUsers, handleChangeColor, interruptRequest } from './rootReducer';
import ProfilePortalBlock from './ProfilePortalBlock';

function Profile(props) {

        const color = useSelector(state => state.color);
        const data = useSelector(state => state.data);
        const error = useSelector(state => state.error);
        const loading = useSelector(state => state.loading);

        const dispatch = useDispatch();

        const inputRef = React.createRef();

        const handleFocus = () => {
            inputRef.current.focus();
        };

        return (
            <>
            <ProfilePortalBlock>
                <div style={{borderColor: color, borderStyle: 'solid'}}>
                    <>
                        <div> request response app! </div>
                        <button style={{borderColor: 'green', color:'green'}} onClick={() => dispatch(handleChangeColor('green'))}>Change border color to green</button>
                        <button style={{borderColor: 'violet', color:'violet'}} onClick={() => dispatch(handleChangeColor('violet'))}>Change border color to violet</button><br/>
                        <button onClick={() => {dispatch(fetchUsers()); handleFocus()}}>Send request to people api</button>
                        <button ref={inputRef} onClick={ () => dispatch(interruptRequest())}>Interrupt Request</button><br />

                        <h3>{loading ? 'Loading user...' : ''}</h3>

                        {data && (
                            <>
                                <img src={data.picture.large} alt='Avatar' />
                                <h2>{data.email}</h2>
                            </>
                        )}
                        {error && (
                            <>
                                <h2>{error}</h2>
                                <button onClick={() => dispatch(fetchUsers())}>Send Request one more time</button><br />
                            </>
                        )}
                    </>
                </div>
            </ProfilePortalBlock>
            </>
        )
}

export default Profile;