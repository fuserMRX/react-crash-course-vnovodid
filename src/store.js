import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

import { usersReducer, userState } from "./rootReducer";

const store = createStore(
    usersReducer,
    userState,
    composeWithDevTools(applyMiddleware(thunk))
);

export default store;
