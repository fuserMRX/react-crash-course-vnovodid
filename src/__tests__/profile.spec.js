import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Profile from '../Profile';
import { Provider } from 'react-redux';
import store from '../store';

describe('<Profile />', () => {
    it('should render and match snapshot', () => {
        const { container } = render(<Provider store={store}><Profile /></Provider>);

        expect(container).toMatchSnapshot();
    })
});