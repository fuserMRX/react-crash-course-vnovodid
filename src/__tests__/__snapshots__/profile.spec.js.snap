// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render and match snapshot 1`] = `
<div>
  <div
    style="border-color: coral; border-style: solid;"
  >
    <div>
       request response app! 
    </div>
    <button
      style="border-color: green; color: green;"
    >
      Change border color to green
    </button>
    <button
      style="border-color: violet; color: violet;"
    >
      Change border color to violet
    </button>
    <br />
    <button>
      Send request to people api
    </button>
    <button>
      Interrupt Request
    </button>
    <br />
    <h3 />
  </div>
</div>
`;
